File: WindowsFormsApp1/Proveedor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Data.SqlClient;

namespace WindowsFormsApp1
{
    public partial class Proveedor : Form
    {
        private SqlConnection connection = Conexion.Connection();
        private Boolean ban = true;
        private String funcion;

        public Proveedor(String funcion)
        {
            this.funcion = funcion;
            InitializeComponent();
            setearBotones();
        }

        private void label3_Click(object sender, EventArgs e)
        {

        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void label6_Click(object sender, EventArgs e)
        {

        }

        private void textBox4_TextChanged(object sender, EventArgs e)
        {

        }

        private void label8_Click(object sender, EventArgs e)
        {

        }

        private void pictureBox3_Click(object sender, EventArgs e)
        {
            MenuProveedores menuProveedores = new MenuProveedores();
            menuProveedores.Show();
            ban = false;
            Close();
        }

        private void Proveedor_Load(object sender, EventArgs e)
        {

        }

        private void Proveedor_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (ban)
            {
                MenuProveedores menuProveedores = new MenuProveedores();
                menuProveedores.Show();
            }
        }

        private void nom_TextChanged(object sender, EventArgs e)
        {

        }

        private void setearBotones()
        {
            switch (funcion)
            {
                case "insertar":
                    insertar.Enabled = true;
                    eliminar.Enabled = false;
                    modificar.Enabled = false;
                    consultar.Enabled = true;
                    clave.Enabled = false;
                    proximaClave();
                    break;
                case "modificar":
                    insertar.Enabled = false;
                    eliminar.Enabled = false;
                    modificar.Enabled = false;
                    consultar.Enabled = true;
                    clave.Enabled = true;
                    nom.Enabled = false;
                    dis.Enabled = false;
                    tel.Enabled = false;
                    dir.Enabled = false;
                    email.Enabled = false;
                    codigo.Enabled = false;
                    ciudad.Enabled = false;
                    clave.Enabled = true;
                    break;
                case "eliminar":
                    insertar.Enabled = false;
                    eliminar.Enabled = false;
                    modificar.Enabled = false;
                    consultar.Enabled = true;
                    nom.Enabled = false;
                    dis.Enabled = false;
                    tel.Enabled = false;
                    dir.Enabled = false;
                    email.Enabled = false;
                    codigo.Enabled = false;
                    ciudad.Enabled = false;
                    clave.Enabled = true;
                    break;
            }
        }

        private void proximaClave()
        {
            int clvProv;
            connection.Open();
            SqlCommand command = new SqlCommand("ULTIMACLVPROVEEDOR", connection);
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.Add("@CLVPROV", SqlDbType.Int).Direction = ParameterDirection.Output;
            command.ExecuteNonQuery();
            clvProv = Convert.ToInt32(command.Parameters["@CLVPROV"].Value.ToString());
            clave.Text = clvProv.ToString();
            connection.Close();
        }

        private void insertar_Click(object sender, EventArgs e)
        {
            Boolean ban = false;
            int error;
            connection.Open();
            SqlCommand command = new SqlCommand("INSERTAPROV", connection);
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.AddWithValue("@NOMPROV", nom.Text.ToUpper());
            command.Parameters.AddWithValue("@NOMDIST", dis.Text.ToUpper());
            command.Parameters.AddWithValue("@TEL", tel.Text.ToUpper());
            command.Parameters.AddWithValue("@DIREC", dir.Text.ToUpper());
            command.Parameters.AddWithValue("@EMAIL", email.Text);
            command.Parameters.AddWithValue("@CP", codigo.Text.ToUpper());
            command.Parameters.AddWithValue("@CIUDAD", ciudad.Text.ToUpper());
            command.Parameters.AddWithValue("@CLVPROV", clave.Text.ToUpper());
            command.Parameters.Add("@BAN", SqlDbType.Int).Direction = ParameterDirection.Output;
            command.ExecuteNonQuery();
            error = int.Parse(command.Parameters["@BAN"].Value.ToString());
            switch (error)
            {
                case 1:
                    MessageBox.Show("Ingrese el nombre del proveedor");
                    break;
                case 2:
                    MessageBox.Show("Ingrese el nombre del distribuidor");
                    break;
                case 3:
                    MessageBox.Show("Ingrese la dirección");
                    break;
                case 4:
                    MessageBox.Show("Ingrese el email");
                    break;
                case 5:
                    MessageBox.Show("Ingrese la ciudad");
                    break;
                default:
                    ban = true;
                    break;
            }
            connection.Close();
            if (ban)
            {
                MessageBox.Show("Proveedor correctamente insertado");
                nom.Text = "";
                dis.Text = "";
                tel.Text = "";
                dir.Text = "";
                email.Text = "";
                codigo.Text = "";
                ciudad.Text = "";
                proximaClave();
            }
        }

        private void consultar_Click(object sender, EventArgs e)
        {
            SqlCommand command;
            SqlDataReader reader;
            connection.Open();
            command = new SqlCommand("SELECT * FROM PROVEEDOR WHERE CLVPROV = " + clave.Text, connection);
            reader = command.ExecuteReader();
            if (reader.Read())
            {
                nom.Text = reader["NOMPROV"].ToString();
                dis.Text = reader["NOMDIST"].ToString();
                tel.Text = reader["TEL"].ToString();
                dir.Text = reader["DIREC"].ToString();
                email.Text = reader["EMAIL"].ToString();
                codigo.Text = reader["CP"].ToString();
                ciudad.Text = reader["CIUDAD"].ToString();
                if (funcion.Equals("modificar"))
                {
                    clave.Enabled = false;
                    nom.Enabled = true;
                    dis.Enabled = true;
                    tel.Enabled = true;
                    dir.Enabled = true;
                    email.Enabled = true;
                    codigo.Enabled = true;
                    ciudad.Enabled = true;
                    modificar.Enabled = true;
                    consultar.Enabled = false;
                }
                if (funcion.Equals("eliminar"))
                {
                    clave.Enabled = false;
                    eliminar.Enabled = true;
                    consultar.Enabled = false;
                }
            }
            else
            {
                MessageBox.Show("No se encontró ningún proveedor con la clave que busca.");
            }
            connection.Close();
        }

        private void eliminar_Click(object sender, EventArgs e)
        {
            Boolean ban = false;
            SqlCommand command;
            connection.Open();
            command = new SqlCommand("ELIMINARPROV", connection);
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.AddWithValue("@CLVPROV", clave.Text);
            command.Parameters.Add("@BAN", SqlDbType.Int).Direction = ParameterDirection.Output;
            command.ExecuteNonQuery();
            switch (Convert.ToInt32(command.Parameters["@BAN"].Value.ToString()))
            {
                case 1:
                    MessageBox.Show("No se encontró ningún proveedor con la clave que busca para eliminar.");
                    break;
                default:
                    ban = true;
                    break;
            }
            if (ban)
            {
                MessageBox.Show("Proveedor eliminado correctamente.");
                nom.Text = "";
                dis.Text = "";
                tel.Text = "";
                dir.Text = "";
                email.Text = "";
                codigo.Text = "";
                ciudad.Text = "";
                clave.Text = "";
                clave.Enabled = true;
                eliminar.Enabled = false;
                consultar.Enabled = true;
            }
            connection.Close();
        }

        private void modificar_Click(object sender, EventArgs e)
        {
            Boolean ban = false;
            SqlCommand command;
            connection.Open();
            command = new SqlCommand("MODIFICAPROV", connection);
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.AddWithValue("@NOMPROV", nom.Text.ToUpper());
            command.Parameters.AddWithValue("@NOMDIST", dis.Text.ToUpper());
            command.Parameters.AddWithValue("@TEL", tel.Text.ToUpper());
            command.Parameters.AddWithValue("@DIREC", dir.Text.ToUpper());
            command.Parameters.AddWithValue("@EMAIL", email.Text);
            command.Parameters.AddWithValue("@CP", codigo.Text.ToUpper());
            command.Parameters.AddWithValue("@CIUDAD", ciudad.Text.ToUpper());
            command.Parameters.AddWithValue("@CLVPROV", clave.Text.ToUpper());
            command.Parameters.Add("@BAN", SqlDbType.Int).Direction = ParameterDirection.Output;
            command.ExecuteNonQuery();
            switch (Convert.ToInt32(command.Parameters["@BAN"].Value.ToString()))
            {
                case 1:
                    MessageBox.Show("Ingrese el nombre del proveedor");
                    break;
                case 2:
                    MessageBox.Show("Ingrese el nombre del distribuidor");
                    break;
                case 3:
                    MessageBox.Show("Ingrese la dirección");
                    break;
                case 4:
                    MessageBox.Show("Ingrese el email");
                    break;
                case 5:
                    MessageBox.Show("Ingrese la ciudad");
                    break;
                case 6:
                    MessageBox.Show("La clave del proveedor no puede quedar vacía");
                    break;
                case 7:
                    MessageBox.Show("La clave del provedor no existe");
                    break;
                default:
                    ban = true;
                    break;
            }
            if (ban)
            {
                MessageBox.Show("Datos del proveedor actualizados");
                nom.Text = "";
                dis.Text = "";
                tel.Text = "";
                dir.Text = "";
                email.Text = "";
                codigo.Text = "";
                ciudad.Text = "";
                nom.Enabled = false;
                dis.Enabled = false;
                tel.Enabled = false;
                dir.Enabled = false;
                email.Enabled = false;
                codigo.Enabled = false;
                ciudad.Enabled = false;
                clave.Enabled = true;
                consultar.Enabled = true;
                modificar.Enabled = false;
            }
            connection.Close();
        }
    }
}

File: Proveedor.vb
Public Class Proveedor
    Private Sub Proveedor_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        menuprin.Enabled = False
        menuprin.Visible = False

        consulta = New ADODB.Recordset
        consulta = conexion.Execute("select clvprov from proveedor order by clvprov")

        While Not consulta.EOF
            CLVPROV.Items.Add(consulta.Fields(0).Value())
            consulta.MoveNext()
        End While

    End Sub

    Private Sub salir_Click(sender As Object, e As EventArgs) Handles salir.Click
        menuprin.Enabled = True
        menuprin.Visible = True
        Close()
    End Sub

    Private Sub consultar_Click(sender As Object, e As EventArgs) Handles consultar.Click
        consulta2 = New ADODB.Recordset
        consulta2 = conexion.Execute("select NOMDIST,NOMPROV,TEL,DIREC,EMAIL,CP,CIUDAD from PROVEEDOR where CLVPROV=" & CLVPROV.Text)

        If Not consulta2.EOF Then
            NOMDIST.Text = consulta2.Fields(0).Value
            NOMPROV.Text = consulta2.Fields(1).Value
            TEL.Text = consulta2.Fields(2).Value
            DIREC.Text = consulta2.Fields(3).Value
            EMAIL.Text = consulta2.Fields(4).Value
            CP.Text = consulta2.Fields(5).Value
            CIUDAD.Text = consulta2.Fields(6).Value

        Else
            MsgBox("LA CLAVE DEL PROVEEDOR NO EXISTE O ESTÁ DADA DE BAJA")
        End If
    End Sub

    Private Sub insertar_Click(sender As Object, e As EventArgs) Handles insertar.Click
        ban = New ADODB.Parameter
        Dim clave As Integer
        clave = 0
        comando = New ADODB.Command
        With comando
            .CommandText = "INSERTAPROV"
            .CommandType = CommandType.StoredProcedure
            .Parameters.Append(.CreateParameter("0", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, NOMDIST.Text))
            .Parameters.Append(.CreateParameter("1", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, NOMPROV.Text))
            .Parameters.Append(.CreateParameter("3", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 12, TEL.Text))
            .Parameters.Append(.CreateParameter("2", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, DIREC.Text))
            .Parameters.Append(.CreateParameter("4", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 35, EMAIL.Text))
            .Parameters.Append(.CreateParameter("5", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamInput, , Val(CP.Text)))
            .Parameters.Append(.CreateParameter("6", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 40, CIUDAD.Text))
            .Parameters.Append(.CreateParameter("7", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamInput, , Val(CLVPROV.Text)))
            .Parameters.Append(.CreateParameter("8", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamOutput, , 0))
            .ActiveConnection = conexion
            .Execute()
            ban.Value = .Parameters(5).Value
        End With
        If ban.Value = 1 Then
            MsgBox("POR FAVOR INSERTA EL NOMBRE DEL PROVEEDOR")
        Else
            If ban.Value = 2 Then
                MsgBox("POR FAVOR INSERTA EL NOMBRE DEL DISTRIBUIDOR")
            Else
                If ban.Value = 3 Then
                    MsgBox("POR FAVOR INSERTA LA DIRECCION")
                Else
                    If ban.Value = 4 Then
                        MsgBox("POR FAVOR INSERTA EL CORREO ELECTRONICO")
                    Else
                        If ban.Value = 5 Then
                            MsgBox("POR FAVOR INSERTA LA CIUDAD DE UBICACION")
                        Else
                            consql = ("select max(clvprov)from proveedor")
                            consulta1 = New ADODB.Recordset
                            consulta1 = conexion.Execute(consql)

                            ' If Not consulta1.EOF Then
                            'clave = consulta1.Fields(0).Value
                        End If

                        MsgBox("INSERCIÓN CORRECTA")
                        NOMDIST.Text = ""
                                    NOMPROV.Text = ""
                                    TEL.Text = ""
                                    DIREC.Text = ""
                                    EMAIL.Text = ""
                                    CP.Text = ""
                                    CIUDAD.Text = ""
                        CLVPROV.Text = ""
                        'clave = 415 + 1

                        'End If
                    End If
                End If
            End If
        End If
    End Sub

    Private Sub eliminar_Click(sender As Object, e As EventArgs) Handles eliminar.Click
        ban = New ADODB.Parameter
        comando = New ADODB.Command
        With comando
            .CommandText = "ELIMPROV"
            .CommandType = CommandType.StoredProcedure
            .Parameters.Append(.CreateParameter("0", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamInput, , Val(CLVPROV.Text)))
            .Parameters.Append(.CreateParameter("1", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamOutput, , 0))
            .ActiveConnection = conexion
            .Execute()
            ban.Value = .Parameters(1).Value
        End With

        If ban.Value = 1 Then
            MsgBox("LA CLAVE DEL PROVEEDOR NO EXISTE")
        Else
            MsgBox("PROVEEDOR ELIMINADO CORRECTAMENTE")
            CLVPROV.Text = ""
            NOMDIST.Text = ""
            NOMPROV.Text = ""
            DIREC.Text = ""
            EMAIL.Text = ""
            TEL.Text = ""
            CP.Text = ""
            CIUDAD.Text = ""

        End If
    End Sub

    Private Sub modificar_Click(sender As Object, e As EventArgs) Handles modificar.Click
        ban = New ADODB.Parameter
        comando = New ADODB.Command
        With comando
            .CommandText = "MODIFICAPROV"
            .CommandType = CommandType.StoredProcedure
            .Parameters.Append(.CreateParameter("0", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, NOMDIST.Text))
            .Parameters.Append(.CreateParameter("1", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, NOMPROV.Text))
            .Parameters.Append(.CreateParameter("3", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 12, TEL.Text))
            .Parameters.Append(.CreateParameter("2", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 50, DIREC.Text))
            .Parameters.Append(.CreateParameter("4", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 35, EMAIL.Text))
            .Parameters.Append(.CreateParameter("5", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamInput, , Val(CP.Text)))
            .Parameters.Append(.CreateParameter("6", DataTypeEnum.adVarChar, ParameterDirectionEnum.adParamInput, 40, CIUDAD.Text))
            .Parameters.Append(.CreateParameter("7", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamInput, , Val(CLVPROV.Text)))
            .Parameters.Append(.CreateParameter("8", DataTypeEnum.adInteger, ParameterDirectionEnum.adParamOutput, , 0))
            .ActiveConnection = conexion
            .Execute()
            ban.Value = .Parameters(7).Value
        End With
        If ban.Value = 1 Then
            MsgBox("POR FAVOR INSERTA EL NOMBRE DEL PROVEEDOR")
        Else
            If ban.Value = 2 Then
                MsgBox("POR FAVOR INSERTA EL NOMBRE DEL DISTRIBUIDOR")
            Else
                If ban.Value = 3 Then
                    MsgBox("POR FAVOR INSERTA LA DIRECCION")
                Else
                    If ban.Value = 4 Then
                        MsgBox("POR FAVOR INSERTA EL CORREO ELECTRONICO")
                    Else
                        If ban.Value = 5 Then
                            MsgBox("POR FAVOR INSERTA LA CIUDAD DE UBICACION")
                        Else
                            If ban.Value = 6 Then
                                MsgBox("LA CLAVE DEL EMPLEADO NO TIENE QUE ESTAR VACIA")
                            Else
                                If ban.Value = 7 Then
                                    MsgBox("LA CLAVE DEL PROVEEDOR NO EXISTE")
                                Else
                                    MsgBox("MODIFICACION REALIZADA CORRECTAMENTE")
                                    NOMDIST.Text = ""
                                    NOMPROV.Text = ""
                                    TEL.Text = ""
                                    DIREC.Text = ""
                                    EMAIL.Text = ""
                                    CP.Text = ""
                                    CIUDAD.Text = ""
                                    CLVPROV.Text = ""

                                End If
                            End If
                        End If
                    End If
                End If
            End If
        End If
    End Sub
End Class